internal/common/baseclient/baseSpeaker.go

```diff
@@ -7,29 +7,48 @@ import (
 type BaseSpeaker struct {
 }
 
-func (s *BaseSpeaker) PayJudge(salary shared.Resources) (shared.Resources, bool) {
-	return salary, true
+// PayJudge is used for paying judge for his service
+func (s *BaseSpeaker) PayJudge(salary shared.Resources) shared.SpeakerReturnContent {
+	return shared.SpeakerReturnContent{
```

Comment:
Apparently, it was to give the code more structure, as for example before it was returning a boolean with no explanation of what it actually meant if it returned true.

Reply:
Could we not have named returns then, I'm not wanting to sound dismissive but I am worried about this disjoint union set up in terms of runtime integrity esp. of functions which clients override, @lhl2617 would you mind weighing in on whether this is ok?

Reply:
if I recall correctly, the previous approach of returning a tuple (value, boolean) was meant to be changed because this return had no meaning. With the return struct, each entry is named and there are checks to check not only the action taken but also that the return was of the expected type (see eg. executive.go, `broadcastTaxation` function)

Reply:
That's fair, but looking at these checks (for example line 44 on legislature.go) we simply get a "Cannot perform ...." error which could have been caused by either the type check or !withdrawSuccess. Looking at the return types we have, I'd say we could benefit from named tuple return types and getting the free type checking that the language provides.

Reply:
That being said I don't want to block this pr over this, so its fine to leave as is for now.